### PR TITLE
feat: support nested describe

### LINKED ISSUE #48: Tests always show as failed in gutter sign and summary even though output shows passed
I just installed and setup for the first time and got things seemingly working except they always show as failed even when the tests are successful. See screenshot. Any thoughts? Thanks!

image
image
image

### PR DESCRIPTION
**Pull request by AbaoFromCUG**
## Features
* Nested test support: (describe>describe>test) #32
*  Vanilla function, `functio(){}` test implement #33
* Add unittest for parse `json reporter's result` to `neotest Tree` 
* Rename `init.vim` to `minimal_init.vim` for simplify unittest with `neotest-plenary`,  `minimal_init.vim` is the default start-up script.

## Fix
* `json reporter`'s output file parse issue #48

### CODE REVIEW

File: lua/neotest-vitest/init.lua [hunk 1]
```diff
@@ -345,7 +304,7 @@ function adapter.build_spec(args)
     "--reporter=verbose",
     "--reporter=json",
     "--outputFile=" .. results_path,
-    "--testNamePattern=" .. testNamePattern,
+    "'--testNamePattern=" .. testNamePattern .. "'",
```
Comment (marilari88) [inline lua/neotest-vitest/init.lua:307]: The single quotes break the test execution in my environment. The whole file tests are executed ignoring the testNamePattern. Is there a special reason you added them?

**Comment by marilari88** [comment]
hi @AbaoFromCUG @joshmedeski Thanks for your huge contribution! Feel free to you take a look at the actions that are failing? in other case I will do as soon as I can

**Comment by AbaoFromCUG** [comment]
I had removed `vim.iter` in `init.lua` because it is new feature of `neovim 0.10`

**Comment by joshmedeski** [comment]
Hey @AbaoFromCUG can you clarify why I'm authored on this commit? I don't remember helping you write any of this code. Maybe it was indirect and I'm missing something.

**Comment by AbaoFromCUG** [comment]
@joshmedeski  I am sorry to bother you,  I guess that the maintainer @marilari88  add you to reviewer.

**Comment by joshmedeski** [comment]
Just to confirm, it says I authored the commit, but I did not. Please check your git config or change the git commit message on this PR so that you are the only author, thanks.

File: lua/neotest-vitest/init.lua (hunk 1 above)
Comment (marilari88) [reply lua/neotest-vitest/init.lua:290]: hey @AbaoFromCUG did you take a look at this comment?

**Comment by marilari88** [comment]
@AbaoFromCUG ping

**Comment by GitMurf** [comment]
Any updates here? Very excited for these changes and fixes!! Thanks!

**Comment by AbaoFromCUG** [comment]
@marilari88 

I have revert the modify about `testNamePattern` single quote issue and resolve the comment.

I test in 3 type configuration manually, and all test cases are passed
* `nested describe`
* `test/it`
* `sub describe`

## 1. Plain Config
```lua
    require("neotest").setup({
        adapters = {
            require("neotest-vitest")({
                vitestCommand = "bunx vitest",
            }),
        }
    })
```
## 2. Your nvim lua config
from `https://github.com/marilari88/nvim-lua-config`
```lua
    require("neotest").setup({
        adapters = {
            require("neotest-vitest")({
                vitestCommand = "bunx vitest",
            }),
        },
        quickfix = { open = true },
        output = {
            enabled = true,
            open = "botright split | resize 15",
        }
    })

```
## 3. My config
`neotest`+`neotest-vitest`+`overseer.nvim`+"toggleterm.nvim`
```lua
     require("overseer").setup({
        templates = { "builtin" },
        strategy = {
            "toggleterm",
            direction = "float",
            use_shell = true,
        },
    })
    ---@diagnostic disable-next-line: duplicate-set-field
    require("overseer.shell").escape_cmd = function(cmd)
        return table.concat(vim.tbl_map(vim.fn.shellescape, cmd), " ")
    end
    require("neotest").setup({
        adapters = {
            require("neotest-vitest")({
                vitestCommand = "bunx vitest",
            }),
        },
        consumers = { overseer = require("neotest.consumers.overseer") },
    })
```

Could you checkout and test it? thanks

**Comment by AbaoFromCUG** [comment]
@marilari88 fix v0.9.5 compatible issue

**Review by marilari88 — APPROVED** [review]

**Comment by GitMurf** [comment]
This is awesome!! Thanks so much for this PR! Quick question to confirm, this does NOT resolve the it.each() dynamic parameterized names issue correct?

**Comment by AbaoFromCUG** [comment]
> This is awesome!! Thanks so much for this PR! Quick question to confirm, this does NOT resolve the it.each() dynamic parameterized names issue correct?

No.

Don't related to `it.each()`/`test.each()`

**Comment by hammerlink** [comment]
@AbaoFromCUG this PR seems to break finding any of the tests for me.
after some research I found out modifying one line in the `discover_positions` function fixes it for me.
```lua
-- before
  query = query .. string.gsub(query, "arrow_function", "function_expression")
--- after
  query = query .. string.gsub(query, "arrow_function", "function")
```
is it just on my treesitter that this issue occurs? Using function_expression in the query gives me 0 results.

**Pull request merged**